plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePlugin.java

```diff
@@ -228,6 +228,89 @@ public void testSystemAccessControl()
+                        .put("hive.metastore.http.client.authentication.type", "BEARER")
+                        .buildOrThrow(),
+                new TestingConnectorContext());
+        connector.shutdown();
+
+        Connector httpConnector = connectorFactory.create(
+                "test",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.metastore.uri", "http://localhost:443")
+                        .put("hive.metastore.http.client.ssl.enabled", "false")
+                        .put("hive.metastore.http.client.additional-headers", "key:value")
+                        .buildOrThrow(),
+                new TestingConnectorContext());
+        httpConnector.shutdown();
+
+        assertThatThrownBy(() -> connectorFactory.create(
+                "test",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.metastore.uri", "http://localhost:443")
+                        .put("hive.metastore.http.client.additional-headers", "key:value")
+                        .put("hive.metastore.http.client.bearer-token", "token")
+                        .put("hive.metastore.http.client.authentication.type", "BEARER")
+                        .buildOrThrow(),
+                new TestingConnectorContext()))
+                .hasMessageContaining("'hive.metastore.http.client.bearer-token' must not be set while using http metastore URIs in 'hive.metastore.uri'");
+
+        assertThatThrownBy(() -> connectorFactory.create(
+                "test",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.metastore.uri", "http://localhost:443, https://localhost:443")
+                        .put("hive.metastore.http.client.ssl.enabled", "true")
+                        .put("hive.metastore.http.client.additional-headers", "key:value")
+                        .put("hive.metastore.http.client.bearer-token", "token")
+                        .buildOrThrow(),
+                new TestingConnectorContext()))
+                .hasMessageContaining("'hive.metastore.uri' cannot contain both http and https URI schemes");
+
+        assertThatThrownBy(() -> connectorFactory.create(
+                "test",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.metastore.uri", "http://localhost:443, thrift://localhost:443")
+                        .put("hive.metastore.http.client.ssl.enabled", "false")
+                        .put("hive.metastore.http.client.additional-headers", "key:value")
+                        .put("hive.metastore.http.client.bearer-token", "token")
+                        .buildOrThrow(),
+                new TestingConnectorContext()))
+                .hasMessageContaining("'hive.metastore.uri' cannot contain both http(s) and thrift URI schemes");
+
+       /* assertThatThrownBy(() -> connectorFactory.create(
```

Comment:
Please don't leave behind commented code.

Reply:
Will do. Right now the PR is only a exploration into the feasibility of the solution.